chips/sam4l/src/dma.rs

```diff
@@ -144,12 +153,18 @@ impl DMAChannel {
         DMAChannel {
             registers: (DMA_BASE_ADDR + (channel as usize) * DMA_CHANNEL_SIZE) as *mut DMARegisters,
             nvic: nvic,
-            client: None,
+            client: Cell::new(None),
+            width: Cell::new(DMAWidth::Width8Bit),
             enabled: Cell::new(false),
             buffer: TakeCell::empty(),
         }
     }
 
+    pub fn initialize(&self, client: &'static mut DMAClient, width: DMAWidth) {
```

Comment:
It could be. However, every `chips` driver has some `set_client` function or another that functions identically to the DMA here (expects to only be called once at initialization time, and drops a reference if called twice).